src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        self.assertIsInstance(megafield_metadata, MegaFieldMetaData)
+
+        # Test attributes megafield_metadata holding only primitive datatypes (int, float, string but not lists)
+        self.assertIsInstance(megafield_metadata.mega_field_id, str)
+        self.assertIsInstance(megafield_metadata.storage_directory, str)
+        self.assertIsInstance(megafield_metadata.custom_data, str)
+        self.assertIsInstance(megafield_metadata.stage_position_x, float)
+        self.assertIsInstance(megafield_metadata.stage_position_x, float)
+        self.assertIsInstance(megafield_metadata.pixel_size, int)
+        self.assertIsInstance(megafield_metadata.dwell_time, int)
+        self.assertIsInstance(megafield_metadata.x_scan_to_acq_delay, int)
+        self.assertIsInstance(megafield_metadata.x_cell_size, int)
+        self.assertIsInstance(megafield_metadata.x_eff_cell_size, int)
+        self.assertIsInstance(megafield_metadata.x_scan_gain, float)
+        self.assertIsInstance(megafield_metadata.x_scan_offset, float)
+        self.assertIsInstance(megafield_metadata.x_descan_offset, int)
+        self.assertIsInstance(megafield_metadata.y_cell_size, int)
+        self.assertIsInstance(megafield_metadata.y_eff_cell_size, int)
+        self.assertIsInstance(megafield_metadata.y_scan_gain, float)
+        self.assertIsInstance(megafield_metadata.y_scan_offset, float)
+        self.assertIsInstance(megafield_metadata.y_descan_offset, int)
+        self.assertIsInstance(megafield_metadata.y_prescan_lines, int)
+        self.assertIsInstance(megafield_metadata.x_scan_delay, int)
+        self.assertIsInstance(megafield_metadata.scan_rotation, float)
+        self.assertIsInstance(megafield_metadata.descan_rotation, float)
+
+        # Test descan setpoints
+        self.assertIsInstance(megafield_metadata.x_descan_setpoints, list)
+        self.assertIsInstance(megafield_metadata.y_descan_setpoints, list)
+        for x_setpoint, y_setpoint in zip(megafield_metadata.x_descan_setpoints, megafield_metadata.y_descan_setpoints):
+            self.assertIsInstance(x_setpoint, int)
+            self.assertIsInstance(y_setpoint, int)
+
+        # Test cell_parameters
+        self.MPPC.cellTranslation.value = \
+            tuple(tuple((10 + j, 20 + j) for j in range(i, i + self.MPPC.shape[0]))
+                  for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+
+        self.MPPC.cellDarkOffset.value = \
+            tuple(tuple(j for j in range(i, i + self.MPPC.shape[0]))
+                  for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+
+        self.MPPC.cellDigitalGain.value = \
+            tuple(tuple(float(j) for j in range(i, i + self.MPPC.shape[0]))
+                  for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+
+        megafield_metadata = self.MPPC._assembleMegafieldMetadata()
```

Comment:
Why do you need to call the assemble method here again. Can you not set the cell parameters at the beginning of the test, then only call once assemble MD and then do all the testing for types etc.?

Reply:
That part lacked a comment, it is because before this I have changed the cell parameters and I want to check if these are represented correctly in the megafieldmetadata object.

Reply:
The reason I want to check with default input before changing any values, is that if there is to be less dependent on other setters etc. Ii the first part of the test. Quite often I did change a VA (or its default value) which created an error. The first part just checks if types are assigned correctly. After that the test gets more complicated and dependent on other parts of the software working correctly.
Note: recalculation of the parameters doesn't cost much time

Reply:
Let's discuss via zoom